rfd/0090-db-mfa-sessions.md

```diff
@@ -0,0 +1,176 @@
+---
+author: Gavin Frazar (gavin.frazar@goteleport.com)
+state: draft
+---
+ 
+# RFD 90 - Database MFA Sessions
+
+## Required Approvers
+- Engineering: Marek `@smallinsky`, Roman `@r0mant`
+- Product: Sasha `@klizhentas`, Xin `@xinding33`
+
+## What
+Use a local proxy tunnel to define the lifetime of a database MFA session, so that database access users are not required to tap an MFA device every minute when per-session-MFA is enabled.
+
+## Why
+Database access via GUI client such as DataGrip requires the user to start a local proxy to connect to: `tsh proxy db --tunnel <db>`.
+
+To support per-session-MFA for the local proxy, we are adding a callback function called on each new connection to the local proxy.
+This callback function will check if the database certs need to be reissued to proxy the connection, and prompt for credentials/MFA if needed.
+
+Currently, database certs are issued with 1 minute ttl if per-session-MFA is enabled as described in [RFD 14](https://github.com/gravitational/teleport/blob/master/rfd/0014-session-2FA.md).
+This causes poor UX when the GUI client can establish a new connection at any time,
+and a user may be prompt once per minute for their MFA.
+
+We should remove the 1 minute restriction on database cert TTL to improve UX.
+
+## Details
+When per-session-MFA is enabled, we should not restrict database cert TTL to 1 minute.
+
+Instead, database cert TTL should be restricted to `max_session_ttl`, and the cert
```

Comment:
Just to clarify - this will only apply to certificates used by `tsh proxy db`, correct? Certificates produced by `tsh db login` (i.e. those that are dumped to disk) will still be valid for 1 minute? Let's call this out explicitly here.

Reply:
Ah, I see below that `tsh db login` will be disabled in per-session MFA mode? Makes sense if so, but let's point it out explicitly here as well.

Reply:
Added to this section that we will disable these commands for per-session-mfa